expo/gbe/models.py

```diff
@@ -395,21 +395,23 @@ def sched_payload(self):
 
     def has_role_in_event(self, role, event):
         '''
-        Gets all of a person's roles for a conference
+        Returns True if this person has the
+        given role in the given event
         '''
         doing_it = False
         if role == "Performer":
             for show in self.get_shows():
                 if show.pk == event.pk:
-                    doing_it = (doing_it or True)
-        else:
-            event_workers = event.roles(role)
-            if self in event_workers:
-                doing_it = True
-            else:
-                for perf in self.get_performers():
-                    doing_it = doing_it or (perf in event_workers)
-
+                    doing_it = True
```

Comment:
Hm. There are definitely better ways to manage complexity - this function should really be about four lines, and call two functions which would handle the two cases - but we'll worry about that when it comes to tackling the overall tech debt. #biggerfishtofry